app/src/main/kotlin/com/quittle/a11yally/preferences/PreferenceProvider.kt

```diff
@@ -203,7 +203,7 @@ class PreferenceProvider(context: Context, resumeOnConstruction: Boolean = false
 /**
  * Hold onto the reference of pref to ensure it doesn't get garbage collected
  */
-internal class PreferenceLiveData<T>(private val pref: PreferenceProviderMember<T>) :
```

Comment:
What needed to access this directly? Is this just to avoid a warning?

Reply:
Yes, this was only to avoid getting a warning.

Reply:
As you can see in the comment above it, while the constructor doesn't think it needs to hold onto it, doing this will result in bugs as the listeners won't get called if the PreferenceProvider gets garbage collected. I actually stumbled into that exact bug a while back. It's confusing because the compiler is perfectly fine with it. The value can get passed in, you can have the live data be referenced by the listener but if you don't keep the listener around, it will get cleaned up even with the live data is still referenced.

You can add the `@Suppress("CanBeParameter")` annotation to suppress this warning.

Reply:
Okay, I should've read that comment more carefully as I mistakenly assumed the job of "holding onto the reference..." was done by the code just after the "private val pref: ..." line. Thank you for the explanation, I understand it now. Sure, I'll add the @Suppress annotation.